### DISCUSSION TITLE
Switching timelines without deleting the dialogic_main_node

### DISCUSSION THREAD

**Discussion by nlupugla** [dialogic-godot/dialogic#1472] [Q&A]
Hi good folks of Dialogic,

I ran into an issue today when trying to switch from one timeline to another. When a timeline ends, Dialogic (I'm using alpha 6) frees the dialogic_main_node. The way I've organized my project, I expected to have a single dialogic_main_node that I was planning to run all my dialogue with. When I try to start a new timeline, the game crashes because dialogic_main_node has been deleted. I thought about simply commenting out the `clear()` line in DialogicGameHandler.end_timeline(), but I have no idea how many other things this might inadvertantly break.

Is there a better way to switch timelines?

Thanks in advance!

**Comment by Jowan-Spooner** [comment]
Hi, this seems to be a bad way to do it indeed. I'm working on a PR that allows choosing how to handle the layout node on timeline end (delete/hide/keep).
So you can comment out that code for now or wait until after the PR/the next alpha release.

**Reply by Jowan-Spooner** [reply]
Hi, as mentioned in other places on main (or in the next alpha release) you can now choose how to treat the layout scene at the end of the dialog. You can choose between hiding, deleting and keeping the scene.
